cli/cli.go

```diff
@@ -0,0 +1,168 @@
+// Package cli provides boilerplate code required to generate plugin executable.
+// At high level it performs following tasks
+// 1. Validate command arguments
+// 2. Read and unmarshal input
+// 3. Execute relevant plugin functions
+// 4. marshals output
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/notaryproject/notation-plugin-framework-go/internal/slices"
+	"github.com/notaryproject/notation-plugin-framework-go/log"
+	"github.com/notaryproject/notation-plugin-framework-go/plugin"
+)
+
+type Cli struct {
```

Comment:
nit: golang convention:
```suggestion
type CLI struct {
```